tools/generate-wire.py

```diff
@@ -384,7 +399,72 @@ def print_fromwire_array(self, ctx, subcalls, basetype, f, name, num_elems):
                 subcalls.append('fromwire_{}(&cursor, &plen, {} + i);'
                                 .format(basetype, name))
 
-    def print_fromwire(self, is_header):
+    def print_tlv_fromwire(self, tlv_name):
+        """ prints fromwire function definition for a TLV message.
+        these are significantly different in that they take in a struct
+        to populate, instead of fields, as well as a length to read in
+        """
```

Comment:
This is an interesting choice, and I'm not sure it's the correct one (we changed from structs to individual parameters early on in our wire parsing).

We could parse TLVs just like we do a formatted field, except everything has to be a ptr-to-ptr so we can set it to NULL if it doesn't appear.

Reply:
right, i think either of these is a valid way to do this. i had the impression that most TLV messages would be rather involved field wise, but it's also possible they're mostly one-off fields...